tools/storyteller/StoryWriter.java

```diff
@@ -18,71 +17,70 @@
-  public static DateFormat DATE_FORMATTER = new SimpleDateFormat("yyyy-MM-dd_HH:mm:ss_z");
+  private static final DateFormat DATE_FORMATTER = new SimpleDateFormat("yyyy-MM-dd_HH:mm:ss_z");
 
   private Config config;
   private FileUtils fileUtils;
+  private StoryList.Builder storiesBuilder;
+  private Story.Builder storyBuilder;
+  private ImmutableList<StoryItem> storyItems;
+  private long lastSavedStoryItemTimeMs;
+  private long startStoryTimeMs;
 
   @Inject
   public StoryWriter(StorytellerConfig storytellerConfig, FileUtils fileUtils) {
     this.config = storytellerConfig.getConfig();
     this.fileUtils = fileUtils;
+    storiesBuilder = StoryList.newBuilder();
+    storyItems = ImmutableList.of();
   }
 
-  private String getSharedStoriesPath() {
-    return Storyteller.getSharedStoriesPath(config);
+  void startStory(String project) {
+    long currentTimeMS = getCurrentTimestamp();
+    startStoryTimeMs = currentTimeMS;
+    lastSavedStoryItemTimeMs = currentTimeMS;
+    buildStory(project);
+    storiesBuilder.addStory(storyBuilder);
+    saveStories();
   }
 
-  private String getUnsharedStoriesPath() {
-    return Storyteller.getUnsharedStoriesPath(config);
-  }
-
-  public void writeStatusFile(FileData.Type type, Message contents) {
-    fileUtils.mkdirs(getUnsharedStoriesPath());
-    // We shift by a second to screenshots will be within START and END.
-    int secondsShift = 0;
-    if (type == FileData.Type.START) {
-      secondsShift = -1;
-    }
-    if (type == FileData.Type.END) {
-      secondsShift = 1;
-    }
-
-    String timeString = getCurrentTimeString(secondsShift);
-    String filename = timeString + "." + type;
-    try {
-      File statusFile = Paths.get(getUnsharedStoriesPath(), filename).toFile();
-      if (contents == null) {
-        statusFile.createNewFile();
-      } else {
-        fileUtils.writePrototxt(contents, statusFile.toString());
-      }
-    } catch (IOException e) {
-      throw new RuntimeException(e);
-    }
-    if (type != FileData.Type.RUNNING) {
-      System.out.println(String.format("\n%s tracking: %s", type, timeString));
-    }
+  void updateStory(String project) {
+    buildStory(project);
+    updateCurrentStory();
+    saveStories();
   }
 
-  private String getCurrentTimeString(int secondsShift) {
-    long currentTime = System.currentTimeMillis() + secondsShift * 1000;
-    return DATE_FORMATTER.format(new Date(currentTime));
+  void endStory(String project) {
+    buildStory(project);
+    updateCurrentStory();
+    saveStories();
+    storyItems = ImmutableList.of();
   }
 
-  private String getCurrentTimeString() {
-    return getCurrentTimeString(0);
+  void saveStoryItem(String project, String story) {
+    // Replace empty with one space, so that the prototxt will have the entry
+    story = story.isEmpty() ? " " : story;
+    long currentTime = getCurrentTimestamp();
+    long timeMs = currentTime - lastSavedStoryItemTimeMs;
```

Comment:
Why are you setting `setTimeMs()` as `currentTime - lastSavedStoryItemTimeMs` and not just `currentTime`?

Reply:
I misunderstood the value of this variable. I thought it was the duration of `item`.